### PR TITLE
Updated wordpress__data-controls to 2.28

### PR DESCRIPTION
**Pull request by marekdedic**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/WordPress/gutenberg/blob/%40wordpress/data-controls%402.28.0/packages/data-controls/CHANGELOG.md
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

I had to remove some tests that depended on types that aren't available anymore. If you have any idea about how to replace them, I'm open to suggestions

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## wordpress__data-controls ([<kbd>unpkg</kbd>](https://unpkg.com/browse/wordpress__data-controls@latest/))
was missing the following properties:
1. syncSelect
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8081f8488b95be7b862e0627e139533c129f5b97
</p>

File: types/wordpress__data-controls/index.d.ts [hunk 1]
```diff
@@ -53,9 +52,9 @@ export function apiFetch(options: APIFetchOptions): unknown;
  * @returns An object for registering the default controls with the store.
  */
 export const controls: {
-    API_FETCH: (action: Action) => Promise<any>;
-    DISPATCH: (action: Action) => void;
-    SELECT: (action: Action) => any;
+    API_FETCH: (action: any) => Promise<any>;
+    DISPATCH: (action: any) => void;
+    SELECT: (action: any) => any;
```
Comment (weswigham) [inline types/wordpress__data-controls/index.d.ts:57]: The shape of this object type [doesn't seem to match the source at all anymore](https://github.com/WordPress/gutenberg/blob/%40wordpress/data-controls%402.28.0/packages/data-controls/src/index.js#L132). Should the available members be resynchronized?

Comment (marekdedic) [reply types/wordpress__data-controls/index.d.ts:57]: Hmm, you're right, however, it seems like it [never did](https://github.com/WordPress/gutenberg/blame/d2e301ed6dd1b0be42f1ca95736b39f735299292/packages/data-controls/src/index.js)... I'll take a look at what I can do with this.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Please handle feedback from @weswigham

File: types/wordpress__data-controls/index.d.ts (hunk 1 above)
Comment (marekdedic) [reply types/wordpress__data-controls/index.d.ts]: Resolved by #65008

**Comment by marekdedic** [comment]
Well, following #65008 this has become a bit of a nothing-burger, but since it at least adds a (albeit deprecated) function, I'll leave it up :)

**Comment by marekdedic** [comment]
Hey @weswigham sorry to mention you directly, but this issue seems like it has nothing to do with the actual changes and is just a manifestation of #24788 - which I can fix manually on my device, but not in the CI... Any ideas on what to do? I could add all dependencies of `@types/react` as dependencies of this package, but that is a dirty solution...

**Comment by marekdedic** [comment]
Obsoleted by #65111.

**Pull request closed without merging**